api/http_client_test_C0_sanitizer_go1_21.go

```diff
@@ -0,0 +1,53 @@
+//go:build !go1.22
+// +build !go1.22
```

Comment:
I couldn't figure out how to specify ﻿< 1.22, so I wrote it like this.

Reply:
I think this is correct but actually I realised this change is not necessary at all because you are bumping the minimum go version in `go.mod` to `1.22` and since `1.21` this `go` version is mandatory i.e. when I'm on `go 1.21` and try to run the tests, it requires `1.22`.

```
➜  cli git:(go-1.22) go test ./...
go: downloading go1.22 (darwin/arm64)
go: download go1.22 for darwin/arm64: toolchain not available
```

With this constraint, we actually don't need to worry about backwards compatibility in the tests so I think we can just remove this file?

Sorry for the extra work!

Reply:
I should have noticed since I was the one making this change, but I didn't look into it properly.

Thanks to this, I was able to learn about the annotation `go:build xxx`